atef/widgets/config.py

```diff
@@ -1209,9 +1211,45 @@ def on_text_changed(self, text: str) -> None:
         # Show or hide the del button as needed
         self.del_button.setVisible(not text)
         # Adjust the width to match the text
-        font_metrics = self.line_edit.fontMetrics()
-        width = font_metrics.boundingRect(text).width()
-        self.line_edit.setFixedWidth(max(width + 10, 40))
+        match_line_edit_text_width(self.line_edit, text=text)
+
+
+def match_line_edit_text_width(
+    line_edit: QLineEdit,
+    text: Optional[str] = None,
+    min: int = 40,
```

Comment:
Just a thought:
Maybe 'minimum width in number of characters based on the metrics' would be a bit more intuitive than dealing in pixels here. I think then it'd be like getting the font metrics of a string with `min` `x`s (`"x" * min`)

Reply:
I was unable to figure out how to do this in terms of character count last time I did this. I could give it another try.

Reply:
I think it's unimportant in the scheme of this PR. To my eyes you have everything you need if you know how to use the character metrics to determine a widget width based on a string, though.

Reply:
I believe you, except that I have not familiarized myself with all of the API and may have made a mistake in not using a monospaced font.